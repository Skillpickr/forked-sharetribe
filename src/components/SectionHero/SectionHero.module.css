@import '../../styles/customMediaQueries.css';

/* Animation for the hero elements */

@-webkit-keyframes animationTitle {
  0% {
    visibility: hidden;
    opacity: 0;
    top: -20px;
  }
  100% {
    visibility: visible;
    opacity: 1;
    top: 0;
  }
}

@keyframes animationTitle {
  0% {
    visibility: hidden;
    opacity: 0;
    top: -20px;
  }
  100% {
    visibility: visible;
    opacity: 1;
    top: 0;
  }
}

/* Shared animation rules */
.animation {
  -webkit-animation-name: animationTitle;
  animation-name: animationTitle;
  -webkit-animation-duration: 0.5s;
  animation-duration: 0.5s;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-animation-fill-mode: forwards;
  -webkit-animation-delay: 3s;
  animation-delay: 3s;

  visibility: hidden;
  opacity: 1;
  position: relative;
}

.root {
  /* composes: defaultBackgroundImage from global; */
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;

  /* Text positioning is done with paddings */
}

.heroContent {
  margin: 0 24px 0 24px;

  @media (--viewportMedium) {
    margin: 0 7.5vw 0 7.5vw;
  }

  /* Special viewport for adjusting the heroContent's placement */

  @media only screen and (min-width: 1025px) and (max-width: 1366px) {
    padding: 0 36px 0 36px;
  }

  @media only screen and (min-width: 1367px) {
    margin: 0 auto;
    max-width: calc(1052px + 36px + 36px);
    width: 100%;
    padding: 0 36px 0 36px;
  }
}
.icon {
  display: inline-block;

  margin: 0 4px;
  text-align: center;
}

.someLinks {
  display: block;
  justify-content: center;
  margin-top: 24px;
  padding-left: 0px;

  @media (--viewportMedium) {
    display: flex;
    padding-left: 8pxpadding-left;
  }
}
.cta {
  display: flex;
  flex-direction: column;
  align-items: center;
  @media (--viewportMedium) {
    flex-direction: row;
    justify-content: center;
  }
}

.heroMainTitle {
  composes: h2 from global;
  color: var(--matterColorDark);
  composes: animation;
  text-align: center;

  @media (--viewportMedium) {
    max-width: var(--SectionHero_desktopTitleMaxWidth);
  }
}
.heroMainTitleFEDelay {
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.heroSubTitle {
  composes: h4 from global;

  color: var(--matterColorLight);
  margin: 0 0 32px 0;
  composes: animation;

  @media (--viewportMedium) {
    max-width: var(--SectionHero_desktopTitleMaxWidth);
    margin: 0 0 47px 0;
  }
}
.heroSubTitleFEDelay {
  -webkit-animation-delay: 0.15s;
  animation-delay: 0.15s;
}

.heroButton {
  composes: button buttonFont buttonText buttonBorders buttonColors from global;
  composes: animation;
  margin-left: auto;
  margin-right: auto;
  margin: 8px;

  @media (--viewportMedium) {
    display: block;
    width: 260px;
  }
}
.heroButtonFEDelay {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}
