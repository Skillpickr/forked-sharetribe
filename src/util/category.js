const Categories = {
  creative: 'creative',
  performance: 'performance',
  knowledge: 'knowledge',
  audioProd: 'audio-prod',
  mediaProd: 'media-prod'
}

const Skills = {
  photographer: 'photographer',
  videographer: 'videographer',
  dj: 'dj',
  musician: 'musician',
  band: 'band',
  makeupArtist: 'makeupArtist',
  dancer: 'dancer',
  miscellaneous: 'miscellaneous'
}
export { Categories, Skills }
