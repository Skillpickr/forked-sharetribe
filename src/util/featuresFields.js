const InputFieldsType = {}

const CheckboxFieldsType = {
  photographerTypeKey: 'photographerType',
  djTypeKey: 'djType',
  musicianTypeKey: 'musicianType',
  musicianGenreKey: 'musicianGenre',
  bandGenreKey: 'bandGenre',
  bandTypeKey: 'bandType'
}

const DropdownFieldsType = {
  musicianSoloKey: 'musicianSoloType',
  ownStudioKey: 'ownStudio',
  soundLightExpKey: 'soundLightExp',
  djGearForPlayingKey: 'djGearForPlaying',
  songRequestKey: 'songRequest'
}

export { InputFieldsType, CheckboxFieldsType, DropdownFieldsType }
