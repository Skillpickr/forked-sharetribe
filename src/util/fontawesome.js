// import the library
import { library, config } from '@fortawesome/fontawesome-svg-core'

// import your icons
import { fas, faCircleQuestion, faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons'

import { fab } from '@fortawesome/free-brands-svg-icons'

config.autoAddCss = false
library.add(
  fas,
  fab,
  faCircleQuestion,
  faMagnifyingGlass
  // more icons go here
)
