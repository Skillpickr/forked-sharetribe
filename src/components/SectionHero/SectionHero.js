import React, { useEffect, useState } from 'react';
import { string } from 'prop-types';
import { FormattedMessage } from '../../util/reactIntl';
import classNames from 'classnames';
import { NamedLink } from '../../components';

import css from './SectionHero.module.css';

const SectionHero = props => {
  const [mounted, setMounted] = useState(false);
  const { rootClassName, className } = props;

  useEffect(() => {
    setMounted(true);
  }, []);

  const classes = classNames(rootClassName || css.root, className);

  return (
    <div className={classes}>
      <div className={css.heroContent}>
        <h1 className={classNames(css.heroMainTitle, { [css.heroMainTitleFEDelay]: mounted })}>
          <FormattedMessage id="SectionHero.title" />
        </h1>
        {/*  <h2 className={classNames(css.heroSubTitle, { [css.heroSubTitleFEDelay]: mounted })}>
          <FormattedMessage id="SectionHero.subTitle" />
        </> */}
        <NamedLink
          name="SearchPage"
          to={{
            search:
              '?address=Denmark&bounds=57.805252798942%2C15.2971743987523%2C54.5011797001343%2C7.9729991066846',
          }}
          className={classNames(css.heroButton, { [css.heroButtonFEDelay]: mounted })}
        >
          <FormattedMessage id="SectionHero.browseButton" />
        </NamedLink>
      </div>
    </div>
  );
};

SectionHero.defaultProps = { rootClassName: null, className: null };

SectionHero.propTypes = {
  rootClassName: string,
  className: string,
};

export default SectionHero;
