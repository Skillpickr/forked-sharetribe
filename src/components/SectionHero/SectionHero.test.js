import React from 'react'
import { renderDeep, renderShallow } from '../../util/test-helpers'
import SectionHero from './SectionHero'
import { userLocation } from '../../util/maps'

const noop = () => null

describe('SectionHero', () => {
  it('matches snapshot', () => {
    window.google = { maps: {} }
    const location = { search: '/s?' }
    const css = ''
    const heroProps = {
      history: { push: noop },
      location: {
        pathname: '/',
        search: '',
        hash: '',
        key: ''
      }
    }
    const tree = renderDeep(<SectionHero {...heroProps} location={location} className={css} />)
    delete window.google
    expect(tree).toMatchSnapshot()
  })
})
