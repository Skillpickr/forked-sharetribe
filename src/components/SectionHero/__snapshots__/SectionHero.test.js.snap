// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHero matches snapshot 1`] = `
<div
  className="root"
>
  <div
    className="heroContent"
  >
    <div
      className=""
    >
      <div
        className="cta"
      >
        <a
          className="heroButton"
          data-tip={true}
          href="/s"
          onClick={[Function]}
          onMouseOver={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          title={null}
        >
          <span>
            SectionHero.browseButton
          </span>
        </a>
        <button
          className="root heroButton"
          disabled={false}
          onClick={[Function]}
        >
          <span>
            SectionHero.introductionVideo
          </span>
        </button>
        <a
          className="heroButton"
          data-tip={true}
          href="/signup"
          onClick={[Function]}
          onMouseOver={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
          title={null}
        >
          <span>
            SectionHero.signUp
          </span>
        </a>
      </div>
    </div>
    <div
      className="someLinks"
    >
      <span>
        SectionHero.followUs
      </span>
      <a
        className="icon"
        href="https://www.instagram.com/skillpickr/"
        rel="noopener noreferrer"
        target="_blank"
        title="Footer.goToInstagram"
      >
        <span>
           Instagram 
        </span>
      </a>
      <span>
        SectionHero.or
      </span>
      <a
        className="icon"
        href="https://www.facebook.com/skillpickr/"
        rel="noopener noreferrer"
        target="_blank"
        title="Footer.goToFacebook"
      >
        <span>
           Facebook 
        </span>
      </a>
      <span>
        SectionHero.latestUpdate
      </span>
    </div>
  </div>
</div>
`;
